Foundation/Process.swift

```diff
@@ -60,6 +60,43 @@ extension CFSocketError {
 }
 #endif
 
+#if !canImport(Darwin) && !os(Windows)
+private func findMaximumOpenFromProcSelfFD() -> CInt? {
+    guard let dirPtr = opendir("/proc/self/fd") else {
+        return nil
+    }
+    defer {
+        closedir(dirPtr)
+    }
+    var highestFDSoFar = CInt(0)
+
+    while let dirEntPtr = readdir(dirPtr) {
+        var entryName = dirEntPtr.pointee.d_name
+        let thisFD = withUnsafeBytes(of: &entryName) { entryNamePtr -> CInt? in
+            CInt(String(decoding: entryNamePtr.prefix(while: { $0 != 0 }), as: Unicode.UTF8.self))
+        }
+        highestFDSoFar = max(thisFD ?? -1, highestFDSoFar)
+    }
+
+    return highestFDSoFar
+}
```

Comment:
Would 
```return try? FileManager.default.contentsOfDirectory(atPath: "/proc/self/fd").compactMap { CInt($0) }.max()```

be a bit simpler?

Reply:
@spevans I wasn't sure about the guarantees around symlinks as in does it follow symlinks, does it not, what happens if the symlink is a broken link, etc.  Therefore I went for `readdir`, happy to change if we think it's fine.